eisen/utils/__init__.py

```diff
@@ -169,19 +169,14 @@ def __init__(self, module, input_names, output_names):
 
         self.module = module
 
-        module_argument_list = inspect.getfullargspec(self.module.forward)[0]
+    def forward(self, *args, **kwargs):
+        input_list = list(args)
 
-        module_argument_list.remove('self')
+        for key in kwargs.keys():
+            if key in self.input_names:
```

Comment:
bug prone. should look like 
```
if key in self.input_names[len(input_list):]
```
 as there is the possibility of additional inputs being passed as positional inputs and taking the place of first few expected named inputs.